test/metabase/db/setup_test.clj

```diff
@@ -36,3 +43,38 @@
         (testing "from a connection URL"
           (test* (mdb.data-source/raw-connection-string->DataSource
                   (str "jdbc:h2:" subname))))))))
+
+(deftest setup-fresh-db-test
+  (mt/test-drivers #{:h2 :mysql :postgres}
+    (testing "can setup a fresh db"
+      (mt/with-temp-empty-app-db [_conn driver/*driver*]
+        (is (= :done
+               (mdb.setup/setup-db! driver/*driver* (mdb.connection/data-source) true)))))))
+
+(deftest setup-a-mb-instance-running-version-lower-than-45
+  (mt/test-drivers #{:h2 :mysql :postgres}
+    (mt/with-temp-empty-app-db [conn driver/*driver*]
+      (with-redefs [liquibase/decide-liquibase-file (fn [& _args] @#'liquibase/changelog-legacy-file)]
+        ;; set up a db in a way we have a MB instance running metabase 42
+        (liquibase/with-liquibase [liquibase conn]
+          (.update liquibase 380 ""))
+        (is (str/starts-with?
+             (t2/select-one-pk (liquibase/changelog-table-name conn) {:order-by [[:dateexecuted :desc]]})
+             "v42")))
+
+      (is (= :done
+             (mdb.setup/setup-db! driver/*driver* (mdb.connection/data-source) true))))))
+
+(deftest setup-a-mb-instance-running-version-greater-than-45
+  (mt/test-drivers #{:h2 :mysql :postgres}
+    (mt/with-temp-empty-app-db [conn driver/*driver*]
+      (with-redefs [liquibase/decide-liquibase-file (fn [& _args] @#'liquibase/changelog-legacy-file)]
+        ;; set up a db in a way we have a MB instance running metabase 45
+        (liquibase/with-liquibase [liquibase conn]
+          (.update liquibase 500 ""))
+        (str/starts-with?
+             (t2/select-one-pk (liquibase/changelog-table-name conn) {:order-by [[:dateexecuted :desc]]})
+             "v45"))
+
+      (is (= :done
+             (mdb.setup/setup-db! driver/*driver* (mdb.connection/data-source) true))))))
```

Comment:
There is a fair amount of repetition here, might want to extract it.

Reply:
how about this ? https://github.com/metabase/metabase/pull/34400/commits/dbedb44dfc63fd23c88d6b728c70d4d19d418c2a